crates/apollo-parser/src/parser/mod.rs

```diff
@@ -107,6 +107,15 @@ impl<'a> Parser<'a> {
         parser
     }
 
+    /// Configure the limit on the number of tokens to parse. If an input document
+    /// is too big, parsing will be aborted.
+    ///
+    /// The default limit is 15 000.
+    pub fn with_token_limit(mut self, token_limit: usize) -> Self {
+        self.lexer = self.lexer.with_limit(token_limit);
+        self
+    }
+
```

Comment:
i feel like the router will probably want to do a recursion limit _and_ token limit. Should we write up an example on how users can set both of these limits in the parser?

Reply:
was wondering if i should change the `with_recursion_limit` to a builder method here, instead of a second constructor?
```rust
Parser::new(input)
  .with_recursion_limit(1_000)
  .with_token_limit(70_000)
```

Reply:
i like a builder as an api for this! maybe even simpler?

```rust
Parser::new(input)
  .recursion_limit(1_000)
  .token_limit(70_000)
```